src/coffea/nanoevents/factory.py

```diff
@@ -141,17 +154,23 @@ def load_buffers(
             f"{start}-{stop}",
         )
         uuidpfn = {partition_key[0]: tree.file.file_path}
-        mapping = UprootSourceMapping(
-            TrivialUprootOpener(uuidpfn, interp_options),
-            start,
-            stop,
-            cache={},
-            access_log=None,
-            use_ak_forth=True,
+        arrays = tree.arrays(
+            keys,
+            entry_start=start,
+            entry_stop=stop,
+            ak_add_doc=interp_options["ak_add_doc"],
             decompression_executor=decompression_executor,
             interpretation_executor=interpretation_executor,
+            how=dict,
         )
-        mapping.preload_column_source(partition_key[0], partition_key[1], tree)
+        source_arrays = {
+            k: _OnlySliceableAs(v, slice(start, stop)) for k, v in arrays.items()
+        }
+        mapping = PreloadedSourceMapping(
+            PreloadedOpener(uuidpfn), start, stop, access_log=None
```

Comment:
Shouldn't this be slicing on, instead, `[0, stop-start]` since we've already sliced the array down to `start:stop` in `tree.arrays` and so any further slicing of the chunk needs to index from zero?

In `preloaded.py` the mapping shouldn't be given a full array usually.

I'm not quite sure how the tests are passing if this is the case though, but this seems off to me.

Reply:
It appears that the preloading mapping is given the full array because in the `extract_column` function we have `return columnhandle[start:stop]` so I just wrapped it to leave the behavior the same. To me I would think we would rather not have that and refactor the preloaded mapping to always assume the arrays are ready to go in the correct shape

Reply:
Oh sorry - I wasn't reading the code right. Yeah your work around is fine. Unless you want to refactor now I'll go ahead and and merge?

Reply:
I'd save the refactor for later since I'd like to remove most of the mapping stuff anyway